engine/classes/Elgg/ViewsService.php

```diff
@@ -363,6 +364,16 @@ public function renderView($view, array $vars = [], $viewtype = '', $issue_missi
 		if (strpos($view, '..') !== false) {
 			return '';
 		}
+		
+		// check for extension deadloops
+		if (!isset($vars[self::EXTENSION_TREE_KEY])) {
```

Comment:
If we can avoid passing this data through $vars, we should. We could add another argument to renderView.